### DISCUSSION TITLE
Interaction between transient and persistent objections is inconsistent (or confusing at best)

### DISCUSSION THREAD

**Discussion by mcclurem** [sqlalchemy/sqlalchemy#7750] [Usage Questions]
### Describe the bug

If you construct a transient object, and add a member relation to a persistent object, that transient object is added to the session but if and only if you have a backref on the relationship. The attached code example can be run twice, toggle the definition of Child.parent relationship declaration according to the comments to show the inconsistency.

### To Reproduce

```python
#!/usr/bin/env python3
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy import create_engine
from sqlalchemy.orm import sessionmaker
from sqlalchemy.schema import UniqueConstraint

engine = create_engine('sqlite:///:memory:', echo=True)
Session = sessionmaker(bind=engine)
session = Session()

Base = declarative_base()

class Parent(Base):
    __tablename__ = "parent"
    id = Column(Integer, primary_key=True)
    name = Column(String)
    children = relationship("Child", back_populates="parent")


class Child(Base):
    __tablename__ = "child"
    __table_args__  = (UniqueConstraint("name"),)
    id = Column(Integer, primary_key=True)
    name = Column(String)
    parent_id = Column(ForeignKey("parent.id"))
    #This version of the relationship declaration fails
    #parent = relationship("Parent", back_populates="children")
    #This version is ok
    parent = relationship("Parent")


Base.metadata.create_all(engine)

parent = Parent(name='mom')
session.add(parent)
child = Child(name='jim')
child.parent = parent
session.add(child)
session.commit()
other_child = Child(name='jim')
other_child.parent = parent
## Based on Child.parent.back_populates or not, this will throw an auto-flush issue
pre_existing_child = session.query(Child).filter(Child.name == other_child.name).first()
```

### Error

```
# Copy complete stack trace and error message here, including SQL log output if applicable.
``` 

### Versions

- OS: Mac 12.3 + Ubuntu 20.04 LTS
- Python: 3.9 and 3.8
- SQLAlchemy: 1.4.31
- Database: Postgres and sqlite confirmed

### Additional context

I think that the change introduced way back in #2372 is really the root of my problem and I cannot locate an argument to return to the "pure" version that existed before that - I always want to explicitly session.add(object) and when I forget to do that I'm ok with the api throwing exceptions. When the magic internals of the api add objects to the database without me doing a session.add that seems like a problem.

I've uncovered this because I added back_populates arguments all over the place now that 1.4.x is warning to add them. I'm making use of webargs/marshmallow_sqlalchemy/etc where the api de-serializes json post requests into an object for me. That object is instantiated as transient and then my code queries to see if it's suitable to add to the database or not before adding it. That query to see if it exists/is appropriate is now causing autoflush exceptions.

Generally I would say that disabling auto-flush is a code smell so I don't want to do that. Manually re-constructing this object is overly verbose and prevents me gaining all the benefits of webargs/marshmallow doing the heavy lifting.

**Comment by CaselIT** [comment]
Hi,

Good news: this behaviour is deprecated and it will be removed on v2.
This behaviour can be customized on the relationship, using [cascade_backrefs](https://docs.sqlalchemy.org/en/14/orm/relationship_api.html?highlight=backref#sqlalchemy.orm.relationship.params.cascade_backrefs). See also https://docs.sqlalchemy.org/en/14/orm/cascades.html#controlling-cascade-on-backrefs if you want to read more

**Comment by zzzeek** [comment]
hey there -

this issue is fixed.  Please see https://docs.sqlalchemy.org/en/14/changelog/migration_14.html#cascade-backrefs-behavior-deprecated-for-removal-in-2-0 

for your test case:

session = Session(future=True)


Cc @ianjosephwilson , re: examples of cascade_backrefs being confusing
